File: WebApplication/src/modules/student/dtos/person-lot.dtos/base.dtos/base-student-doctoral.dto.ts
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto"
import { InstitutionDto } from "../helpers/insitution.dto";
import { BasePeDto } from "./base-pe.dto";

export class BaseStudentDoctoralDto extends BasePeDto {
  institution: InstitutionDto;
  subordinate: NomenclatureDto;
  speciality: NomenclatureDto;
  admissionReason: NomenclatureDto;
  educationalForm: NomenclatureDto;
  educationalQualification: NomenclatureDto;
  duration: number | null;
  studentStatus: NomenclatureDto;
  studentEvent: NomenclatureDto;
}

File: StudentCard.Application/Students/Dtos/PersonLotDtos/BaseDtos/BaseStudentDoctoralDto.cs
using StudentCard.Infrastructure.Helpers.Dtos;

namespace StudentCard.Application.Students.Dtos.PersonLotDtos.BaseDtos
{
    public class BaseStudentDoctoralDto : BasePeDto
    {
        public NomenclatureDto Institution { get; set; }

        public NomenclatureDto Subordinate { get; set; }

        public NomenclatureDto Speciality { get; set; }

        public NomenclatureDto AdmissionReason { get; set; }

        public NomenclatureDto EducationalForm { get; set; }

        public NomenclatureDto EducationalQualification { get; set; }

        public NomenclatureDto StudentStatus { get; set; }

        public NomenclatureDto StudentEvent { get; set; }

        public decimal? Duration { get; set; }
    }
}
